taskcluster/client.py

```diff
@@ -400,11 +400,23 @@ def _makeHttpRequest(self, method, route, payload):
                 )
 
             # Handle non 2xx status code and retry if possible
+            status = response.status_code
             try:
                 response.raise_for_status()
-                if response.status_code == 204:
+                if status == 204:
                     return None
 
+                # raise_for_status only catches status >= 400, but it also
+                # customizes the error message.  Let's use that customized
+                # error message for status >= 400, and craft one for other
+                # non-2xx statuses.
+                if status < 200 or status >= 300:
+                    raise requests.exceptions.RequestException(
```

Comment:
Something like this is what I'm referring to, similar to the changes in async.  I just don't really want the control flow to differ that much between sync and async when it doesn't have to.

```diff
diff --git a/taskcluster/client.py b/taskcluster/client.py
index 08bd2dc..23a715d 100644
--- a/taskcluster/client.py
+++ b/taskcluster/client.py
@@ -408,57 +408,57 @@ class BaseClient(object):
 
             # Handle non 2xx status code and retry if possible
             status = response.status_code
-            try:
-                response.raise_for_status()
-                if status == 204:
-                    return None
-
-                # raise_for_status only catches status >= 400, but it also
-                # customizes the error message.  Let's use that customized
-                # error message for status >= 400, and craft one for other
-                # non-2xx statuses.
-                if status < 200 or status >= 300:
-                    raise requests.exceptions.RequestException(
-                        '%s Error: %s for url: %s' % (
-                            response.status_code, response.reason, response.url
-                        ), response=response
-                    )
+            if status == 204:
+                return None
 
-            except requests.exceptions.RequestException as rerr:
-                status = response.status_code
-                if 500 <= status and status < 600 and retry < retries:
-                    log.warn('Retrying because of: %s' % rerr)
-                    continue
-                # Parse messages from errors
-                data = {}
-                try:
-                    data = response.json()
-                except:
-                    pass  # Ignore JSON errors in error messages
-                # Find error message
-                message = "Unknown Server Error"
-                if isinstance(data, dict):
-                    message = data.get('message')
-                else:
-                    if status == 401:
-                        message = "Authentication Error"
-                    elif status == 500:
-                        message = "Internal Server Error"
-                # Raise TaskclusterAuthFailure if this is an auth issue
-                if status == 401:
-                    raise exceptions.TaskclusterAuthFailure(
-                        message,
-                        status_code=status,
-                        body=data,
-                        superExc=rerr
-                    )
-                # Raise TaskclusterRestFailure for all other issues
+            # Catch retryable errors and go to the beginning of the loop
+            # to do the retry
+            if 500 <= status and status < 600 and retry < retries:
+                log.warn('Retrying because of a %s status code' % status)
+                continue
+
+            # Throw errors for non-retryable errors
+            if status < 200 or status >= 300:
                 raise exceptions.TaskclusterRestFailure(
+                    '%s Error: %s for url: %s' % (
+                        response.status_code, response.reason, response.url
+                    ), 
+                    status_code=status,
+                    response=response,
+                    body=data,
+                    superExc=None,
+                )
+
+            # Handle valid responses
+            data = {}
+            try:
+                data = response.json()
+            except:
+                pass  # Ignore JSON errors in error messages
+            # Find error message
+            message = "Unknown Server Error"
+            if isinstance(data, dict):
+                message = data.get('message')
+            else:
+                if status == 401:
+                    message = "Authentication Error"
+                elif status == 500:
+                    message = "Internal Server Error"
+            # Raise TaskclusterAuthFailure if this is an auth issue
+            if status == 401:
+                raise exceptions.TaskclusterAuthFailure(
                     message,
                     status_code=status,
                     body=data,
-                    superExc=rerr
+                    superExc=None
                 )
+            # Raise TaskclusterRestFailure for all other issues
+            raise exceptions.TaskclusterRestFailure(
+                message,
+                status_code=status,
+                body=data,
+                superExc=None
+            )
 
             # Try to load JSON
             try:
```